libraries/ESP8266WiFi/src/ESP8266WiFiSTA.cpp

```diff
@@ -455,10 +455,9 @@ int8_t ESP8266WiFiSTAClass::waitForConnectResult(unsigned long timeoutLength) {
     if((wifi_get_opmode() & 1) == 0) {
         return WL_DISCONNECTED;
     }
-    using esp8266::polledTimeout::oneShot;
-    oneShot timeout(timeoutLength); // number of milliseconds to wait before returning timeout error
+    using oneShotYieldMs = esp8266::polledTimeout::timeoutTemplate<false, esp8266::polledTimeout::YieldPolicy::YieldOrSkip>;

```

Comment:
I know @devyte loves his templates, but this is one confusing template change.  Looking at the implementation it seems correct, but man...does it feel like its obfuscating things.

Reply:
I actually didn't include that premade type for exactly this reason. However, it's a valid use (creating a new polledTimeout variation), and @dok-net isn't tjw first to realize that this particular variation is useful, if a bit... sophisticated. 
It was part of the design choice: if you want such variations, you can use them in exactly this way.

Reply:
I don't share your concern about the 10000µs, because it's open to optimization case-by-case. Changing it globally on the contrary looks dangerous to me.
As a compromise, given that 10000 does seem commonplace, could you agree to making that a default argument value for `optimistic_yield`? It's just gut feeling, but 10ms seems just right for a simple decimal duration.

Reply:
I don't share your concern about the 10000µs, this way it's open to optimization or fixing case-by-case. Introducing a common global on the contrary looks dangerous to me if it gets changed. Unfortunately, default argument isn't available due to extern "C" linkage.